js/viewer/annotation/text.js

```diff
@@ -0,0 +1,425 @@
+function Text(attrs) {
+
+    attrs = attrs ? attrs : {};
+    Base.call(this, attrs);
+
+    this._tag = "text";
+    this._attrs["x"] = (attrs["x"] / 2);
+    this._attrs["y"] = (attrs["y"] / 2);
+    this._attrs["stroke"] = attrs["stroke"] || "red";
+
+    // Calculate the ratio for the mapping from image to screen size
+    var viewerParent = this.parent.parent.parent;
+    var containerSize = viewerParent.osd.viewport.containerSize;
+    this._ratio = Math.max(attrs["x"]/containerSize.x, attrs["y"]/containerSize.y);
+
+    this._attrs["font-size"] = attrs["font-size"] || this._ratio * 14;
+    this._attrs["font-weight"] = attrs["font-weight"] || 400;
+    this.svg = null;
+
+    // Foreign object that contains the annotation input and transformed text p tag
+    this.foreignObj = null;
+    this.setForeignObj = function(obj) {
+        this.foreignObj = obj;
+    }
+    this.getForeignObj = function() {
+        return this.foreignObj;
+    }
+}
+
+Text.prototype = Object.create(Base.prototype);
+Text.prototype.constructor = Text;
+
+/**
+ * This function is used to place the annotation at the desired position when user clicks on the OSD canvas
+ * @param {Array} point - The x and y coordinates of the point are used to position the annotation
+ */
+Text.prototype.positionAnnotation = function (point) {
+    var _self = this;
+    var updateAttrs = {
+        x: point.x,
+        y: point.y,
+    };
+
+    _self.setAttributesByJSON(updateAttrs);
+    _self.renderSVG();
+};
+      
+
+
+/**
+ * Function to transform the <textarea> input to <p> tag when annotation tool is switched.
+ */
+Text.prototype.transform = function () {
+
+    // Transform only if the input is not empty
+    if(textInput.value && textInput.value.trim() != "")
+    {
+        var pText = this.createPTag(textInput);
+        var foreignObj = this.getForeignObj();
+        var div = foreignObj.childNodes[0];
+        var h = div.style.top;
+        var w = div.style.left;
+        foreignObj.style.width = pText.style.width;
+        foreignObj.style.height = pText.style.height;
+        var divPadding = window.getComputedStyle(div, null).getPropertyValue('padding');
+        divPadding = parseInt(divPadding.slice(0, divPadding.length - 2)) || 0;
```

Comment:
Since we are defining the styling while creating the div when the text annotation is created. We can ensure that the sizing will always be in pixels. Commented the same in the code.